### PR TITLE
Add collector service

### PR DESCRIPTION
**Pull request by CarlosSilva-zd2**
This pull request brings you the fantastic functionality of collecting all your app usage data, activity, location, name and much more. 
Still needs a button to dump all the usage data, but I need a place to put the button. When the place is available I shall implement this functionality.

### CODE REVIEW

**Review by hgg — CHANGES_REQUESTED** [review]
Why are you uploading the .jar file for the csv lib you are using? Isn't there a way to import it through Gradle?
Something like http://opencsv.sourceforge.net/dependency-info.html ?

File: app/src/main/java/corp/kairos/adamastor/collector/CollectorService.java [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+            case AudioManager.RINGER_MODE_SILENT : result = 1; break;
+            case AudioManager.RINGER_MODE_VIBRATE : result = 2; break;
+            default : result = 0;
+        }
+
+        return result;
+    }
+
+
+    private synchronized boolean getAndLockCycle(){
```
Comment (hgg) [inline app/src/main/java/corp/kairos/adamastor/collector/CollectorService.java]: I'm missing the point of the purpose of this method and the `inCycle` instance variable. Won't the method always return true? Why are you synchronizing this if only this thread is accessing the variable?

**Comment by FranciscoRibeiro** [comment]
I can't test it now, but it looks equivalent to what we had with the previous version.
If anyone needs these changes to work on something, I would say to go ahead and merge.

File: app/src/main/java/corp/kairos/adamastor/collector/CollectorService.java (hunk 1 above)
Comment (CarlosSilva-zd2) [reply app/src/main/java/corp/kairos/adamastor/collector/CollectorService.java]: You are right. This was an artifact from before we figured out how to stop Android from creating multiple instances of the same Service. I removed the variable and method.

**Pull request merged**